src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    public String delete(int taskNumber, Storage fo) {
+        if (taskNumber < 0 || taskNumber >= count()) {
+            return "Invalid Task Number!";
+        } else {
+            Task deletedTask = tasks.remove(taskNumber);
+            fo.writeAllTasksToFile(this);
+            StringBuilder sb = new StringBuilder();
+            sb.append("Noted. I have removed the following task:\n");
+            sb.append("\t" + deletedTask + "\n");
+            sb.append(getCountInWords());
+            return sb.toString();
+        }
+    }
+
+    public int count() {
+        return tasks.size();
+    }
+
+    public Task get(int taskNumber) {
```

Comment:
Consider renaming to something like getTaskAtIndex/getTask in line with getters.